hnn_core/viz.py

```diff
@@ -836,8 +836,41 @@ def plot_connectivity_matrix(net, conn_idx, ax=None, show_weight=True,
     return ax.get_figure()
 
 
-def plot_cell_connectivity(net, conn_idx, src_gid, ax=None, colorbar=True,
-                           colormap='viridis', show=True):
+def _update_target_plot(ax, conn, src_gid, src_type_pos, target_type_pos,
+                        src_range, target_range, nc_dict, colormap):
+    from .cell import _get_gaussian_connection
+
+    # Extract indeces to get position in network
+    # Index in gid range aligns with net.pos_dict
+    target_src_pair = conn['gid_pairs'][src_gid]
+    target_indeces = np.where(np.in1d(target_range, target_src_pair))[0]
+
+    src_idx = np.where(src_range == src_gid)[0][0]
+    src_pos = src_type_pos[src_idx]
+
+    # Aggregate positions and weight of each connected target
+    weights, target_x_pos, target_y_pos = list(), list(), list()
+    for target_idx in target_indeces:
+        target_pos = target_type_pos[target_idx]
+        target_x_pos.append(target_pos[0])
+        target_y_pos.append(target_pos[1])
+        weight, _ = _get_gaussian_connection(src_pos, target_pos, nc_dict)
+        weights.append(weight)
+
+    ax.clear()
+    im = ax.scatter(target_x_pos, target_y_pos, c=weights, s=50,
+                    cmap=colormap)
+    x_pos = target_type_pos[:, 0]
+    y_pos = target_type_pos[:, 1]
+    ax.scatter(x_pos, y_pos, color='k', marker='x', zorder=-1, s=20)
+    ax.scatter(src_pos[0], src_pos[1], marker='s', color='red', s=150)
+    ax.set_ylabel('Y Position')
+    ax.set_xlabel('X Position')
+    return im
+
+
+def plot_cell_connectivity(net, conn_idx, src_gid=None, axes=None,
```

Comment:
Update docstring to indicate interactive plot? Also it'd be good to add a notes section describing how to use it.